javascript/javascript3/week3/ScreenShots/script.js

```diff
@@ -0,0 +1,254 @@
+// initializing variables
+// Decided not to hide crudcrudKEY anywhere - it is valid for only 24 hours
+const screenshotKEY = "af35d9bda9msh1d5d601753355fcp1d5ca1jsnc37e78a50099";
+const crudcrudKEY = "3454dc8fc29e4c269600bb3028ff235f";
+const crudcrudURL = "https://crudcrud.com/api";
+const baseSCR_URL = "https://website-screenshot6.p.rapidapi.com/screenshot";
+let currentUser = {
+  email: "john@smith",
+  hash: -946852072
+};
+
+/**
+ * Deletes the screenshot record from crudcrud.
+ * @param {hex} id - unique number to identify user. the format provided by crudcrud.com
+ */
+async function deleteScreenShot(id) {
+  const response = await fetch(
+    `${crudcrudURL}/${crudcrudKEY}/screenshots/${id}`,
+    {
+      method: "DELETE"
+    }
+  );
+  console.dir(response.status);
+  document.getElementById(id).remove();
+}
+
+/**
+ * Gets the target url, then calls website-screenshot6.p.rapidapi.com to get the link with the image.
+ * Posts the screenshot and metadata to crudcrud by calling postScreenshot.
+ * Then calls renderScreenshot to show it on the page.
+ * @param {object} event - form submit.
+ */
+async function getScreenShot(event) {
+  event.preventDefault();
+  const targetSite = document.getElementById("target-address").value;
+
+  const buttonElement = document.getElementById("get-screenshot-button");
+
+  const options = {
+    method: "GET",
+    headers: {
+      "X-RapidAPI-Key": screenshotKEY,
+      "X-RapidAPI-Host": "website-screenshot6.p.rapidapi.com"
+    }
+  };
+
+  buttonElement.innerText = "Fetching your url...";
+
+  try {
+    const response = await fetch(
+      `${baseSCR_URL}?url=${targetSite}&width=1920&height=1080`,
+      options
+    );
+    if (!response.ok) {
+      throw new Error(response.message);
+    }
+    buttonElement.innerText = "Prepairing the picture for display...";
+    const result = await response.json();
+
+    const screenshotUrl = result.screenshotUrl;
+    document.getElementById("target-address").value = "";
+    const screeenshotToRender = await postScreenshot(
+      screenshotUrl,
+      targetSite,
+      currentUser
+    );
+
+    renderScreenshot(screeenshotToRender);
+    buttonElement.innerText = "Get your screenshot";
+  } catch (error) {
+    screenshotElement.innerText = `Cannot fetch your url, please check its validity, error: ${error.message}`;
+  }
+}
+
+/**
+ * Returns a hash code from a string - to emulate secure storage of passwords.
```

Comment:
This method is impressive!

Reply:
Must admit I just googled for more or less decent replacement for a hash function to hide user passwords. Not my invention :)